01_fib/patrickmarabeas/main_test.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := fmt.Sprintf("%s\n",
+		"0\n1\n1\n2\n3\n5\n8\nERROR: 100 is outside of limits Please enter a number between -92 and 92")
+	got := buf.String()
+
+	t.Run("Main function", func(t *testing.T) {
+		if expected != got {
+			t.Errorf("\nExpected: %s\nGot:      %s", expected, got)
+		}
+	})
+}
+
+func TestCalculateFibSequence(t *testing.T) {
+	var cases = map[string]struct {
+		input     int
+		expected1 []int
+		expected2 error
+	}{
+		"Zero": {
+			input:     0,
+			expected1: nil,
+			expected2: nil,
+		},
+		"Input of 1": {
+			input:     1,
+			expected1: []int{0},
+			expected2: nil,
+		},
+		"Input of negative 1": {
+			input:     -1,
+			expected1: []int{0},
+			expected2: nil,
+		},
+		"Input of 3": {
+			input:     3,
+			expected1: []int{0, 1, 1},
+			expected2: nil,
+		},
+		"Input of 7": {
+			input:     3,
+			expected1: []int{0, 1, 1, 2, 3, 5, 8},
+			expected2: nil,
+		},
+		"Input of negative 3": {
+			input:     -3,
+			expected1: []int{0, -1, -1},
+			expected2: nil,
+		},
+		"Input of negative 7": {
+			input:     3,
+			expected1: []int{0, -1, -1, -2, -3, -5, -8},
+			expected2: nil,
+		},
+		"Out of range - positive": {
+			input:     100,
+			expected1: nil,
+			expected2: errors.New("ERROR: 100 is outside of limits Please enter a number between -92 and 92"),
+		},
+		"Out of range - negative": {
+			input:     -100,
+			expected1: nil,
+			expected2: errors.New("ERROR: -100 is outside of limits Please enter a number between -92 and 92"),
+		},
+	}
+
+	for name, c := range cases {
+		got1, got2 := calculateFibSequence(c.input)
+		expected1 := c.expected1
+		expected2 := c.expected2
```

Comment:
These three lines belong in the test that is run by the runner. It keeps the test case together. You'll probably need a `c := c` in the loop before the `t.Run` call, due to the loop variable being re-used and `t.Run` running concurrently.

Reply:
Resolved